tests/components/minio/test_minio.py

```diff
@@ -0,0 +1,195 @@
+"""Tests for Minio Hass related code."""
+import asyncio
+import json
+from unittest.mock import MagicMock, patch
+
+from homeassistant.components.minio import QueueListener, DOMAIN, CONF_HOST, \
+    CONF_PORT, CONF_ACCESS_KEY, CONF_SECRET_KEY, CONF_SECURE, CONF_LISTEN, \
+    CONF_LISTEN_BUCKET
+from homeassistant.const import EVENT_HOMEASSISTANT_START, \
+    EVENT_HOMEASSISTANT_STOP
+from homeassistant.core import callback
+from homeassistant.setup import async_setup_component
+
+from tests.components.minio.common import TEST_EVENT
+
+
+async def test_minio_services(hass, caplog):
+    """Test Minio services."""
+    hass.config.whitelist_external_dirs = set('/tmp')
+
+    minio_client_mock = MagicMock()
+
+    with patch('homeassistant.components.minio.Minio') as minio_mock:
```

Comment:
I would break this out into a pytest fixture function in which we yield the client mock.